prover/src/trace/trace_lde/default/mod.rs

```diff
@@ -28,20 +28,30 @@ mod tests;
 ///   will always be elements in the base field (even when an extension field is used).
 /// - Auxiliary segments: a list of 0 or more segments for traces generated after the prover
 ///   commits to the first trace segment. Currently, at most 1 auxiliary segment is possible.
-pub struct DefaultTraceLde<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>> {
+pub struct DefaultTraceLde<
+    E: FieldElement,
+    H: ElementHasher<BaseField = E::BaseField>,
+    V: VectorCommitment<H>,
+> {
     // low-degree extension of the main segment of the trace
     main_segment_lde: RowMatrix<E::BaseField>,
     // commitment to the main segment of the trace
-    main_segment_tree: MerkleTree<H>,
+    main_segment_vector_com: V,
     // low-degree extensions of the auxiliary segment of the trace
     aux_segment_lde: Option<RowMatrix<E>>,
     // commitment to the auxiliary segment of the trace
-    aux_segment_tree: Option<MerkleTree<H>>,
+    aux_segment_vector_com: Option<V>,
```

Comment:
I don't love using `*_vector_com` here because we are overloading the term "commitment" to mean both to the commitment "string" and commitment "object" (with actual data). Maybe "oracle" would be a better term - but, I'm honestly not sure.

Reply:
Yeah, it is indeed a bit confusing, although I tried to add `vector` to differentiate the object from the string.
We can use oracle as well, it is not a bad choice either